pkg/sosreport/index.js

```diff
@@ -61,18 +61,31 @@ function sos_create() {
     //        an API exists.
 
     var output = "";
-    var progress_regex = /Running ([0-9]+)\/([0-9]+):/g;
+    var plugins_count = 0;
+    var finishing_regex = /Finishing plugins.*\[Running: (.*)\]/g;
+    var progress_regex = /Starting ([0-9]+)\/([0-9]+).*\[Running: (.*)\]/g;
     var archive_regex = /Your sosreport has been generated and saved in:[ \r\n]+(\/[^\r\n]+)/;
 
     task.stream(function (text) {
         if (sos_task == task) {
             var m, p;
+            p = 0;
 
             output += text;
-
-            p = 0;
-            while ((m = progress_regex.exec(output))) {
-                p = (parseInt(m[1], 10) / parseInt(m[2], 10)) * 100;
+            var lines = output.split("\n");
+            for (var i = lines.length - 1; i >= 0; i--) {
```

Comment:
Going over the lines in reverse and stopping at the first match is more efficient, but not strictly totally correct: If we get the whole output in one go, the loop would see a "Finishing" line while plugins_count is still zero.  But in that case, we didn't report incremental progress either, so we could just set p to 100.